### PR TITLE
Switch ArrayList usage to List

### PR DESCRIPTION
**Pull request by Therzok**

### CODE REVIEW

**Comment by Therzok** [comment]
This should improve some performance on enumeration and ToArray calls.

**Comment by Therzok** [comment]
@slluis PTAL

File: Mono.Addins/Mono.Addins.Description/AddinDescription.cs [hunk 1]
```diff
@@ -1392,7 +1393,7 @@ void IBinaryXmlElement.Read (BinaryXmlReader reader)
 			nodeSets = (ExtensionNodeSetCollection) reader.ReadValue ("NodeSets", new ExtensionNodeSetCollection (this));
 			extensionPoints = (ExtensionPointCollection) reader.ReadValue ("ExtensionPoints", new ExtensionPointCollection (this));
 			conditionTypes = (ConditionTypeDescriptionCollection) reader.ReadValue ("ConditionTypes", new ConditionTypeDescriptionCollection (this));
-			fileInfo = (object[]) reader.ReadValue ("FilesInfo", null);
+			fileInfo = (AddinFileInfo[]) reader.ReadValue ("FilesInfo", null);
```
Comment (slluis) [inline Mono.Addins/Mono.Addins.Description/AddinDescription.cs]: This is not going to work. ReadValue can't infer the array type.

Comment (Therzok) [reply Mono.Addins/Mono.Addins.Description/AddinDescription.cs]: Cool, thanks for the catch! I'll fix that

Comment (Therzok) [reply Mono.Addins/Mono.Addins.Description/AddinDescription.cs]: But even if it's just object, `(AddinFileInfo[])(object[])` should just work

**Comment by Therzok** [comment]
After fixing the requested changes, it works fine in a smoke test.

**Pull request merged**